lib/node/httpStack.js

```diff
@@ -5,22 +5,44 @@ import {Readable, Transform} from "stream";
 import throttle from "lodash.throttle";
 
 export default class NodeHttpStack {
+  constructor(requestOptions = {}) {
+    this._requestOptions = requestOptions;
+  }
+
   createRequest(method, url) {
-    return new Request(method, url);
+    return new Request(method, url, this._requestOptions);
   }
 
   getName() {
     return "NodeHttpStack";
+    // return this.constructor.name;
+  }
+}
+
+/**
+ * A NodeHttpStack which will allow developer to customer `http(s).request()`
+ * options, yet force it to accept insecure certificates as well.
+ */
+export class InsecureNodeHttpStack extends NodeHttpStack {
```

Comment:
Frankly, I don't think we need to export an InsecureNodeHttpStack if the users can simply pass their own options to NodeHttpStack. That just increases the API surface with little benefit IMO.

Reply:
OK, I can leave with this :)